bpfd/src/server/command.rs

```diff
@@ -0,0 +1,86 @@
+//! Commands between the RPC thread and the BPF thread
+
+use bpfd_api::ParseError;
+use tokio::sync::oneshot;
+use uuid::Uuid;
+
+use crate::server::errors::BpfdError;
+
+/// Provided by the requester and used by the manager task to send
+/// the command response back to the requester.
+type Responder<T> = oneshot::Sender<T>;
+
+/// Multiple different commands are multiplexed over a single channel.
+#[derive(Debug)]
+pub(crate) enum Command {
+    /// Load a program
+    Load {
+        path: String,
+        section_name: String,
+        program_type: ProgramType,
+        attach_type: AttachType,
+        username: String,
+        responder: Responder<Result<Uuid, BpfdError>>,
+    },
+    Unload {
```

Comment:
I think we're going to want a ProgramType in both the Unload and List commands.

Reply:
Good point. I also need to adjust the protobuf definition for ListResponse.
But first I'm going to work through the internal changes required as a result of adding tracepoints, which should (hopefully) inform the design of the ListReponse RPC.
It's likely that some sort of AttachInfo will need to be modelled per-program type but I'm finding things out as I go